src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -325,10 +373,90 @@ internal static bool ValidateParentTagName(string parentTag, ErrorSink errorSink
+
+                if (attribute[attribute.Length - 1] != ']')
+                {
+                    errorSink.OnError(SourceLocation.Zero, "TODO: Create error for invalid CSS Selector syntax", length: 0);
+                    return false;
+                }
+
+                var equalsIndex = attribute.IndexOf('=');
+                if (equalsIndex != -1)
+                {
+                    var emptyValueSelector = false;
+
+                    // Don't need to bounds check here because the ']' error validation ensures that there's at least 1
+                    // extra character after the '='.
+                    var afterEquals = attribute[equalsIndex + 1];
+                    if ((afterEquals == '\'' || afterEquals == '"'))
+                    {
+                        // Check if there's a separate matching quote at the end of the attribute selector.
+                        if (attribute.Length - 2 != equalsIndex + 1 &&
+                            attribute[attribute.Length - 2] != afterEquals)
+                        {
+                            errorSink.OnError(SourceLocation.Zero, "TODO: Create error for unbalanced quotes for CSS Selector.", length: 0);
+                            return false;
+                        }
+                        else if (equalsIndex + 2 == attribute.Length - 2)
+                        {
+                            emptyValueSelector = true;
+                        }
+                    }
+                    else if (equalsIndex + 1 == attribute.Length -1)
+                    {
+                        emptyValueSelector = true;
+                    }
+
+                    if (emptyValueSelector)
+                    {
+                        errorSink.OnError(SourceLocation.Zero, "TODO: Create error for empty attribute selector value.", length: 0);
+                        return false;
+                    }
+                }
+
+                attribute = GetCSSAttributeName(attribute, equalsIndex);
+            }
+            else if (attribute[attribute.Length - 1] == RequiredAttributeWildcardSuffix)
+            {
+                // A single '*' at the end of a required attribute is valid; everywhere else is invalid. Strip it from
+                // the end so we can validate the rest of the name.
+                attribute = attribute.Substring(0, attribute.Length - 1);
+            }
+
+            return ValidateName(attribute, targetingAttributes: true, errorSink: errorSink);
+        }
+
+
+        private static string GetCSSAttributeName(string selector, int equalsIndex)
+        {
+            string name;
+            if (equalsIndex == -1)
+            {
+                Debug.Assert(
+                    selector[selector.Length - 1] == ']',
+                    "Validation should have ensured we're never handed an invalid CSS selector.");
+
+                name = selector.Substring(1, selector.Length - 2);
+
+                return name;
```

Comment:
Suspect `[ name ]` is also valid according to the CSS spec. If so, need `.Trim()` here.